code/controllers/subsystem/dcs.dm

```diff
@@ -7,10 +7,22 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
 /datum/controller/subsystem/processing/dcs/Recover()
 	comp_lookup = SSdcs.comp_lookup
 
-/datum/controller/subsystem/processing/dcs/proc/GetElement(eletype)
-	. = elements_by_type[eletype]
+/datum/controller/subsystem/processing/dcs/proc/GetElement(datum/element/eletype, ...)
+	var/element_id = eletype
+	
+	if(initial(eletype.element_flags) & ELEMENT_BESPOKE)
+		var/list/fullid = list("[eletype]")
+		for(var/i in initial(eletype.id_arg_index) to length(args))
+			var/argument = args[i]
+			if(istext(argument) || isnum(argument))
+				fullid += "[argument]"
+			else
+				fullid += "[REF(argument)]"
```

Comment:
Mayhaps some recursion for lists as arguments. Alternatively one could just add a comment that ELEMENT_BESPOKE does not support list arguments.

Reply:
At least I think this won't work with lists.

Reply:
You can get a ref for lists. I'm intentionally leaving this barebones and I'm not going to expand it's functionality more here.

Reply:
Oh... you can get ref for lists? That's neat. Good to know.

Reply:
You can get ref for anything, even numbers

Reply:
Add some mention to the docs it uses ref so it's obvious if anyone tries passing in lists or similar.